overlord/snapstate/snapmgr.go

```diff
@@ -391,8 +393,29 @@ func refreshScheduleNoWeekdays(rs []*timeutil.Schedule) error {
 }
 
 func (m *SnapManager) checkRefreshSchedule() ([]*timeutil.Schedule, error) {
-	refreshScheduleStr := defaultRefreshSchedule
+	if RefreshScheduleManaged != nil {
+		if RefreshScheduleManaged(m.state) {
+			if m.currentRefreshSchedule != "managed" {
+				logger.Noticef("refresh.schedule is managed via the snapd-control interface")
+				m.currentRefreshSchedule = "managed"
+			}
+			// check for refresh hints from the store every 24h
+			// to be able to warn in the future
+			var lastRefreshHints time.Time
+			if err := m.state.Get("last-refresh-hints", &lastRefreshHints); err != nil && err != state.ErrNoState {
```

Comment:
two comments,
1) as far as I understand  we always want to do this  (because the schedule can be slower than once a day even in other cases)
2) this is a strange place to do this anyway in terms of sane error reporting among other things

Reply:
Thanks, I misunderstood and was not aware of (1). Fixed now. And agreed to (2), I moved it but want to look at this a bit harder to see if the existing code can be written in a slightly cleaner way (its a bit entangled right now).